TCDT Milestone 2

khalidzahra opened the pull request:
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] This pull request is being made by the same account as the accepted application.
- [x] I have disclosed any and all sources of reused code in the submitted repositories and have done my due diligence to meet its license requirements.
- [x] In case of acceptance, invoices must be submitted and payments will be transferred to the Polkadot AssetHub and/or fiat account provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://grants.web3.foundation/docs/Support%20Docs/milestone-deliverables-guidelines).

Link to the application pull request: Private application. Project name: tcdt

semuelle reviewed (changes requested):
Hi @khalidzahra, the tcdt-milestone-1.md file is already merged (https://github.com/w3f/Grant-Milestone-Delivery/pull/1241). Can you remove it from this PR?

Also, sorry for the delay. @PieWol is currently out, but will soon return to evaluating.

khalidzahra commented:
Hey @semuelle, no worries! I removed the tcdt-milestone-1.md file as you requested :)

PieWol commented:
Hey @khalidzahra ,
sorry for the delay. I was out last week. Could you please adjust the links for the protocol deliverables so that they point to the relevant code? I'd argue that would be more expressive than the images within the readme. What do you think?

khalidzahra commented:
Hey @PieWol, thank you for your valuable feedback and sorry for the delayed response. I have created testing guides (linked at the bottom of the main README.md of the project) for the milestone's deliverables, and I went ahead and added unit tests that can simply be run to verify their functionality. 

Regarding the protocols, some parts are designed to be offloaded to other blockchains that want to use the project, so there wouldn't be any implementation present in the repository to point to. Furthermore, the revocation protocol will be fully implemented in the third milestone, so I will update the links to point to it then. I have however added links to the relevant code snippets that contribute to each protocol under their respective sections within the README.md.

Let me know if anything is missing, and if these changes are what you expected!

khalidzahra commented:
Hey @PieWol, sorry for the delay! I have added scripts to help in launching the architecture and to also set it up with some data for testing purposes. I updated the testing guides to include how the architecture can be launched as well. Let me know if this is what you were looking for!

semuelle commented:
Hey @khalidzahra, sorry for the wait. I will take over the evaluation for Piet while he's out. I'll get back to you with feedback as soon as possible.

PieWol commented:
Hey @khalidzahra ,
thanks for the updates. I'm finally back from vacation and took a look already. Sadly the unit tests are currently failing. I updated my [evaluation](https://github.com/w3f/Grant-Milestone-Delivery/pull/1258) accordingly. Let me know whats wrong. Once this is resolved I'll proceed with the actual testing with a deployed node.

khalidzahra commented:
Hey @PieWol, thanks for catching that. There was some mismatch between module names which was causing the tests to fail, but I have fixed it and everything should run smoothly.

khalidzahra commented:
Hey @PieWol, I ran the scripts and they seem to work just fine on my end. The only thing that you might be missing is that you must build the node first using cargo before running the scripts. Can you confirm that you have done that?

PieWol commented:
Hey @khalidzahra ,
I did build the node, strictly following your written instructions. Yet even though I now tried the whole process on a local machine to outrule any firewall issues, I still get the same error.

PieWol commented:
I'm assuming that building the node and then executing the ``init.go`` script will take care of everything. Is there anything else that needs to be done prior to executing the script?

khalidzahra commented:
Hi @PieWol, sorry about all the trouble with running the script. I believe the problem probably lies in the dependencies of the DNS client not being installed. I pushed an update that makes init.go take care of it, but if you want to use the version that you currently have, you can simply run `npm install` in the `dns_client` directory before running init.go.

PieWol commented:
I gave it a fresh setup after your commit but it still fails. Manually executing ``npm install`` in the dns_client directory doesn't help either.

khalidzahra commented:
Let's hop into a quick meeting. I will book a slot on your calendar.

PieWol commented:
Thanks, all working!

PieWol commented:
Happy to let you know that the delivery is now accepted. Please invoice the foundation as defined in our private contract.
